src/lib.rs

```diff
@@ -48,19 +49,26 @@ pub mod random;
 #[no_mangle]
 pub extern "C" fn panic_fmt(fmt: core::fmt::Arguments, file: &'static str, line: u32) -> ! {
     use core::fmt::Write;
+    use interrupts::primask_mutex::PrimaskMutex;
 
-    hprintln_err!("\nPANIC in {} at line {}:", file, line);
-    hprintln_err!("    {}", fmt);
+    // Disable all interrupts after panic
+    let mutex: PrimaskMutex<()> = PrimaskMutex::new(());
+    mutex.lock(|_| {
+        hprintln_err!("\nPANIC in {} at line {}:", file, line);
+        hprintln_err!("    {}", fmt);
 
-    unsafe { lcd::stdout::force_unlock() }
-    lcd::stdout::with_stdout(|stdout| {
-        if let Some(ref mut stdout) = *stdout {
-            let _ = writeln!(stdout, "\nPANIC in {} at line {}:", file, line);
-            let _ = writeln!(stdout, "    {}", fmt);
-        }
+        unsafe { lcd::stdout::force_unlock() }
+        lcd::stdout::with_stdout(|stdout| {
+            if let Some(ref mut stdout) = *stdout {
+                let _ = writeln!(stdout, "\nPANIC in {} at line {}:", file, line);
+                let _ = writeln!(stdout, "    {}", fmt);
+            }
+        });
+
+        loop {}
     });
 
-    loop {}
+    unreachable!("Lock will end in a endless loop!");
```

Comment:
you can get rid of this by having `PrimaskMutex::lock` return the value of its closure.